github/users_projects.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// ListProjects lists the projects for an user.
+//
+// GitHub API docs: https://developer.github.com/v3/projects/#list-user-projects
+func (s *UsersService) ListProjects(ctx context.Context, user string, opt *ProjectListOptions) ([]*Project, *Response, error) {
+	u := fmt.Sprintf("users/%v/projects", user)
+	u, err := addOptions(u, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: remove custom Accept header when this API fully launches.
+	req.Header.Set("Accept", mediaTypeProjectsPreview)
+
+	var projects []*Project
+	resp, err := s.client.Do(ctx, req, &projects)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return projects, resp, nil
+}
+
+// CreateProject creates a GitHub Project for the specified user.
+//
+// GitHub API docs: https://developer.github.com/v3/projects/#create-a-user-project
+func (s *UsersService) CreateProject(ctx context.Context, user string, opt *ProjectOptions) (*Project, *Response, error) {
```

Comment:
I think a new struct needs to be created here because the parameters are not the same for this endpoint.  Something like:

```
// CreateUserProjectOptions specifies the parameters to the UsersService.CreateProject method.
type CreateUserProjectOptions {
	// The name of the project. (Required.)
	Name string `json:"name"`
	// The description of the project. (Optional.)
	Body *string `json:"body,omitempty"`
}
```

Reply:
@gmlewis  

thank you for your feedback!

I found that `RepositoriesService` and `OrganizationsService` use  `ProjectOptions` (ref: https://github.com/google/go-github/blob/9a2656da257b4c8208d61ee971374b502a6be4b1/github/projects.go#L64-L87).

`ProjectOptions` includes optional fields which `UsersService.CreateProject` does not need it. But, I use `ProjectOptions` to make the code consistent.

Reply:
Unfortunately, it is no longer acceptable to send extra parameters to GitHub endpoints (I don't have time right now to point you to the other relevant issues, but they are closed in this repo and can be searched for), so we have taken the stance in this repo that we will not allow the uninformed user of this client library to mistakenly send extra parameters.  Therefore, the cleanest option is to create a new options struct.  Please do so.

Reply:
updated: https://github.com/google/go-github/pull/1295/commits/02658a3d6da0288de76057533a934bffc02d9b39